File: sdks/python/bolt_prediction_sdk/models/carbon_model_config_update_request.py
"""
CarbonModelConfigUpdateRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class CarbonModelConfigUpdateRequest(SDKBaseModel):
    """碳排模型系数配置更新请求"""

    degradation: Optional[Any] = Field(description="预紧力劣化模型参数（可选更新）", default=None)
    leakage: Optional[Any] = Field(description="泄漏率估算模型参数（可选更新）", default=None)
    energy_carbon: Optional[Any] = Field(description="能耗与碳排模型参数（可选更新）", default=None)
    operator_id: Optional[Any] = Field(description="操作人ID", default=None)
    operator_name: Optional[Any] = Field(description="操作人姓名", default=None)
    description: Optional[Any] = Field(description="变更说明", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/CarbonModelConfigUpdateRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 碳排模型系数配置更新请求 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class CarbonModelConfigUpdateRequest {

    @JsonProperty("degradation")
    private Object degradation;

    @JsonProperty("leakage")
    private Object leakage;

    @JsonProperty("energy_carbon")
    private Object energyCarbon;

    @JsonProperty("operator_id")
    private Object operatorId;

    @JsonProperty("operator_name")
    private Object operatorName;

    @JsonProperty("description")
    private Object description;

    public Object getDegradation() {
        return degradation;
    }

    public void setDegradation(Object degradation) {
        this.degradation = degradation;
    }

    public Object getLeakage() {
        return leakage;
    }

    public void setLeakage(Object leakage) {
        this.leakage = leakage;
    }

    public Object getEnergyCarbon() {
        return energyCarbon;
    }

    public void setEnergyCarbon(Object energyCarbon) {
        this.energyCarbon = energyCarbon;
    }

    public Object getOperatorId() {
        return operatorId;
    }

    public void setOperatorId(Object operatorId) {
        this.operatorId = operatorId;
    }

    public Object getOperatorName() {
        return operatorName;
    }

    public void setOperatorName(Object operatorName) {
        this.operatorName = operatorName;
    }

    public Object getDescription() {
        return description;
    }

    public void setDescription(Object description) {
        this.description = description;
    }

}